src/batch/src/Job/Item/Processor/ChainProcessor.php

```diff
@@ -59,10 +37,8 @@ public function process($item)
     /**
      * @inheritDoc
      */
-    public function flush(): void
+    protected function getDecoratedElements(): iterable
     {
-        foreach ($this->processors as $processor) {
-            $this->flushElement($processor);
-        }
+        return $this->processors;
```

Comment:
Why not `yield from $this->processors;`?

Reply:
Function accept any iterable, an array is iterable.

What will be the gain in wrapping this array into a Generator ?

Reply:
Won't it free memory? I guess the gain is not huge here, but any performance improvement is good :slightly_smiling_face:

Reply:
I don't think so, processors is already an array in use, so it has to be in memory.

Generator are only relevant when ALL the data flow is using an iterator.

Reply:
All right then :slightly_smiling_face: